Thanks to sydarn2 for this patch to fix rga scaling for non-rotated screened rk devices

diff --git a/gfx/drivers_context/drm_go2_ctx.c b/gfx/drivers_context/drm_go2_ctx.c
index b6e50073f4..f2d3917126 100644
--- a/gfx/drivers_context/drm_go2_ctx.c
+++ b/gfx/drivers_context/drm_go2_ctx.c
@@ -118,20 +118,11 @@ static void *gfx_ctx_go2_drm_init(void *video_driver)
 
    drm->display       = go2_display_create();
 
-   drm->native_width  = go2_display_height_get(drm->display);
-   drm->native_height = go2_display_width_get(drm->display);
-
-   /* This driver should only be used on rotated screens */
-   if (drm->native_width < drm->native_height)
-   {
-      /* This should be fixed by using wayland/weston... */
-      go2_display_destroy(drm->display);
-      free(drm);
-      return NULL;
-   }
+   drm->native_width  = go2_display_width_get(drm->display);
+   drm->native_height = go2_display_height_get(drm->display);
 
    drm->presenter     = go2_presenter_create(drm->display,
-         DRM_FORMAT_RGB565, 0xff000000, true);
+         DRM_FORMAT_XRGB8888, 0xff000000, true);
 
    return drm;
 }
@@ -347,8 +338,8 @@ static void gfx_ctx_go2_drm_swap_buffers(void *data)
    go2_presenter_post(drm->presenter,
          surf,
          src_x, src_y, src_w, src_h,
-         out_y, out_x, out_h, out_w,
-         GO2_ROTATION_DEGREES_270, 2);
+         out_x, out_y, out_w, out_h,
+         GO2_ROTATION_DEGREES_0, 2);
    go2_context_surface_unlock(drm->context, surf);
 #endif
 }
